### ISSUE TITLE
Separate API unit tests from integration tests

### ISSUE THREAD

**Issue by JohnKallies** [wgu-opensource/osmt#42]
tldr;
The Dockerfile has a ```mvn clean package``` that skips tests. We need to adjust the test context in the API module to not skip unit tests. 

### Details
The Kotlin JUnit tests in the api module are a mixture of both unit tests and integration tests. The integration tests stand up test containers. These can not be run from the Dockerfile, which is build during a docker-compose.

This issue requires configuring both Maven Surefire and Failsafe to run the right tests at the appropriate Maven lifecycle phase.
- Unit tests should be run during the "test" phase, using the Surefire plugin, as currently configured 
- Integration tests should be run during the "Verify" phase using the Failsafe plugin. Integration test classes should be renamed to end with "IT" instead of "Test".
- Reinspect the unit test coverage after these changes are made, and improve it if needed.
